frontend/src/app/App.test.tsx

```diff
@@ -917,6 +917,51 @@ describe("App.onHistoryChange", () => {
       expect(instance.onPageChange).toHaveBeenLastCalledWith("sub_hash")
     })
   })
+
+  it("doesn't rerun when we are on the same page and the url contains an anchor", () => {
+    const pushStateSpy = jest.spyOn(window.history, "pushState")
+
+    const instance = wrapper.instance() as App
+    window.history.pushState({}, "", "#foo_bar")
+    jest.spyOn(instance, "onPageChange")
+    instance.onHistoryChange()
+
+    // check that onPageChange is not run when anchor is in the url and we are on the same page
+    expect(instance.onPageChange).not.toHaveBeenCalled()
+    // @ts-expect-error
+    window.history.pushState.mockClear()
+    pushStateSpy.mockRestore()
```

Comment:
nit: (both here and below)

Is it sufficient to just call `pushStateSpy.mockRestore()`? According to the documentation, `mockRestore` already does all of the things that both `mockClear` and `mockReset` do + some more